src/Security/Authentication/BearerToken/src/BearerTokenConfigureOptions.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authentication.BearerToken;
+
+internal sealed class BearerTokenConfigureOptions(IDataProtectionProvider dp) : IConfigureNamedOptions<BearerTokenOptions>
+{
+    private const string _primaryPurpose = "Microsoft.AspNetCore.Authentication.BearerToken";
+
+    public void Configure(string? schemeName, BearerTokenOptions options)
+    {
+        if (schemeName is null)
+        {
+            return;
+        }
+
+        options.BearerTokenProtector = new TicketDataFormat(dp.CreateProtector(_primaryPurpose, schemeName, "BearerToken"));
+        options.RefreshTokenProtector = new TicketDataFormat(dp.CreateProtector(_primaryPurpose, schemeName, "RefreshToken"));
```

Comment:
You don't need a separate data protector, you can pass a purpose directly to Un/Protect. Or is this just to make it so the refresh endpoint doesn't need to know the purpose?

Reply:
This mainly so the refresh endpoint doesn't need to know the purpose. I thought another protector was easier to use than having to know what constant to look for.